nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py

```diff
@@ -746,13 +746,32 @@ def _greedy_decode_blank_as_pad_loop_labels(
         The main idea: search for next labels for the whole batch (evaluating Joint)
         and thus always evaluate prediction network with maximum possible batch size
         """
+        prev_labels = None
+        prev_state = None
         if partial_hypotheses is not None:
-            raise NotImplementedError("`partial_hypotheses` support is not implemented")
+            prev_labels = torch.tensor(
+                [hyp.y_sequence[-1] if len(hyp.y_sequence) else self._blank_index for hyp in partial_hypotheses]
+            ).to(device=x.device)
+
+            state_list = [hyp.dec_state for hyp in partial_hypotheses]
+            if all(state is not None for state in state_list):
+                prev_state = self.decoder.batch_unsplit_states(
+                    [hyp.dec_state for hyp in partial_hypotheses], device=x.device, dtype=x.dtype
+                )
+            elif any(state is not None for state in state_list):
+                logging.warning(f"State list: {state_list}")
+                raise NotImplementedError
 
-        batched_hyps, alignments, last_decoder_state = self._decoding_computer(x=x, out_len=out_len)
+        batched_hyps, alignments, last_decoder_state = self._decoding_computer(
+            x=x, out_len=out_len, prev_labels=prev_labels, prev_state=prev_state
+        )
         hyps = rnnt_utils.batched_hyps_to_hypotheses(batched_hyps, alignments, batch_size=x.shape[0])
         for hyp, state in zip(hyps, self.decoder.batch_split_states(last_decoder_state)):
             hyp.dec_state = state
+
+        if partial_hypotheses:
+            for prev_hyp, hyp in zip(partial_hypotheses, hyps):
+                hyp.y_sequence = torch.cat((prev_hyp.y_sequence, hyp.y_sequence))
```

Comment:
I don't believe this is enough, if we want the results to match.

This is what I did for CTC (WIP!)

```
+            if previous_hypotheses is not None:                                                                                                                                                            
+                for i, hypothesis in enumerate(hypotheses_list):                                                                                                                                           
+                    # I am mutating previous_hypothses here... Perhaps not the wisest idea                                                                                                                 
+                    previous_hypotheses[i].merge(hypothesis)                                                                                                                                               
+                hypotheses_list = previous_hypotheses                                                                                                                                                      
```

This is the implementation of merge():

```
modified   nemo/collections/asr/parts/utils/rnnt_utils.py
@@ -139,6 +139,27 @@ class Hypothesis:                                                                                                                                                                      
         """                                                                                                                                                                                                
         return [] if self.text is None else self.text.split()                                                                                                                                              
                                                                                                                                                                                                            
+    # TODO: Does htis need to run inside of torch.inference_mode() or torch.no_grad()?                                                                                                                     
+    def merge(self, other):                                                                                                                                                                                
+        self.score += other.score                                                                                                                                                                          
+        # TODO: Consider what to do if this is a tensor, not a list. Concatenate?                                                                                                                          
+        self.y_sequence.extend(other.y_sequence)                                                                                                                                                           
+        self.dec_state = other.dec_state                                                                                                                                                                   
+        if self.timestep is not None:                                                                                                                                                                      
+            self.timestep.extend(other.timestep)                                                                                                                                                           
+        self.length += other.length                                                                                                                                                                        
+        self.last_token = other.last_token                                                                                                                                                                 
+                                                                                                                                                                                                           
+        # TODO: Concatenate for alignments and frame_confidence.                                                                                                                                           
+        if self.alignments is not None:                                                                                                                                                                    
+            self.alignments[0] = torch.cat(self.alignments[0], other.alignments[0])                                                                                                                        
+            self.alignments[1] = torch.cat(self.alignments[1], other.alignments[1])                                                                                                                        
+        if self.frame_confidence is not None:                                                                                                                                                              
+            self.frame_confidence.extend(other.frame_confidence)                                                                                                                                           
+                                                                                                                                                                                                           
+        # Invalidated. Need to rerun decode_hypothesis here.                                                                                                                                               
+        self.text = None                                                                                                                                                                                   
+                                                                                                                                                                                                           
                                                                                                                                                                                                            
 @dataclass                                                                                                                                                                                                 
 class NBestHypotheses:                                                                                                                                                                                     
```